backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -938,6 +952,18 @@ protected void legacyRBACDeleteEventTypeFromEndpoint(final SecurityContext secur
     private void internalDeleteEventTypeFromEndpoint(final SecurityContext securityContext, final UUID eventTypeId, final UUID endpointId) {
         final String orgId = getOrgId(securityContext);
         endpointEventTypeRepository.deleteEndpointFromEventType(eventTypeId, endpointId, orgId);
+
+        // Sync behavior group model
+        List<BehaviorGroup> behaviorGroupsLinkedToThisEndpoint = behaviorGroupRepository.findBehaviorGroupsByEndpointId(orgId, endpointId);
+        for (BehaviorGroup behaviorGroup : behaviorGroupsLinkedToThisEndpoint) {
+            if (!behaviorGroup.getBehaviors().isEmpty() && behaviorGroup.getBehaviors().stream().anyMatch(bg -> bg.getId().eventTypeId.equals(eventTypeId))) {
```

Comment:
This isn't really needed:
```suggestion
            if (behaviorGroup.getBehaviors().stream().anyMatch(bg -> bg.getId().eventTypeId.equals(eventTypeId))) {
```